test/App/Fossa/Config/Utils.hs

```diff
@@ -0,0 +1,44 @@
+{-# LANGUAGE TemplateHaskell #-}
+
+module App.Fossa.Config.Utils (parseArgString, itShouldLoadFromTheConfiguredBaseDir) where
+
+import App.Fossa.Config.ConfigFile (ConfigFile (..))
+import Control.Algebra (Has)
+import Control.Effect.Lift (Lift, sendIO)
+import Data.ByteString qualified as BS
+import Options.Applicative (Parser, execParserPure, handleParseResult, header, info, prefs)
+import Path (mkRelFile, toFilePath, (</>))
+import Test.Effect (EffectStack, it', shouldBe', withTempDir)
+import Test.Hspec (Spec)
+
+-- | Parses an arg string or raises an error
+parseArgString :: (Has (Lift IO) sig m) => Parser a -> String -> m a
+parseArgString parser = sendIO . handleParseResult . execParserPure (prefs mempty) (info parser progInfo) . words
+  where
+    progInfo =
+      header "Test Arg Parser"
+
+-- | Tests that the config loader uses the directory set in the arguments
+itShouldLoadFromTheConfiguredBaseDir ::
+  Parser opts -> (opts -> EffectStack (Maybe ConfigFile)) -> Spec
+itShouldLoadFromTheConfiguredBaseDir parser loadConfig =
+  it' "should load from the configured base dir"
+    . withTempDir "AnalyzeSpec"
+    $ \tempDir ->
+      do
+        sendIO $ BS.writeFile (toFilePath (tempDir </> $(mkRelFile ".fossa.yml"))) "version: 42\n"
```

Comment:
[suggestion]: if you use `version: 3`, you'll be able to use `mempty` for the expected config file value.

Reply:
I want to be sure that we don't also default the version because we can't find the config.  I want to be sure that the version came from the file I created so I set it to an unusual value.

Reply:
We don't have a default config file, we handle a `Maybe ConfigFile` everywhere

Reply:
This is not a big deal, just making clear that this suggestion would be semantically safe.

Reply:
That's true.  I think this makes it abundantly clear that there's no way it's falling back to defaults so it's less likely the test could give a false positive.

Reply:
In a way what I want to test is that the config file is read, not that the config file matches a default, so I think this better describes the test.

We should have some tests about the default value though. :D

Reply:
If you just want to test that the file is read, you should use `isJust` instead.  That way the value doesn't matter at all.  As I said, we NEVER fallback to a default config file.